### ISSUE TITLE
Unexpected keyword argument with ParamSpec

### ISSUE THREAD

**Issue by cdce8p** [microsoft/pyright#3160]
**To Reproduce**
```py
from typing import Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")

def run(f: Callable[P, None], *args: P.args, **kwargs: P.kwargs) -> None:
    ...

def g(func: Callable[[int, str], None]) -> None:
    run(func, 2, msg="Hello")


"""
test.py:6: note: "run" defined here
test.py:10: error: Unexpected keyword argument "msg" for "run"
"""
```

**Expected behavior**
`run` excepts `**kwargs`. Furthermore the parameter types match `[int, str]` from `func`. Thus no error should be emitted.

**VS Code extension or command-line**
pyright 1.1.227

**Comment by JelleZijlstra** [comment]
Pyright is correct here. When you type the kwargs as `P.kwargs`, it means the function takes exactly the kwargs that are provided by P. In this case, P is solved to `[int, str]`, which is two positional-only arguments.

You should be able to fix this by using a callable protocol instead of `Callable[[int, str], None]`.
